torch/csrc/jit/script/sugared_value.cpp

```diff
@@ -363,9 +395,26 @@ RangeValue::RangeValue(
     throw ErrorReport(loc) << "range expected at most 3 arguments, got "
                            << inputs.size();
   }
+
+  if (static_len) {
+    static_len_ = static_len;
+  } else {
+    auto len_node = len(loc, m)->node();
+    auto maybe_out_stack = tryConstantPropNode(len_node);
```

Comment:
In person: let's not allow constant prop for ranges unless they are constructed with a static_len